pkg/clients/repositorysecrets/repositorysecrets.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repositorysecrets
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/crossplane-contrib/provider-github/apis/repositorysecrets/v1alpha1"
+	ghclient "github.com/crossplane-contrib/provider-github/pkg/clients"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-github/v33/github"
+	"golang.org/x/crypto/nacl/box"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Service defines the Secrets operations
+type Service interface {
+	GetRepoPublicKey(ctx context.Context, owner, repo string) (*github.PublicKey, *github.Response, error)
+	GetRepoSecret(ctx context.Context, owner, repo, name string) (*github.Secret, *github.Response, error)
+	CreateOrUpdateRepoSecret(ctx context.Context, owner, repo string, eSecret *github.EncryptedSecret) (*github.Response, error)
+	DeleteRepoSecret(ctx context.Context, owner, repo, name string) (*github.Response, error)
+}
+
+// NewService creates a new Service based on the *github.Client
+// returned by the NewClient SDK method.
+func NewService(token string) *Service {
+	c := ghclient.NewClient(token)
+	r := Service(c.Actions)
+	return &r
+}
+
+// CreateOrUpdateSec create or update repository secret in GitHub
+func CreateOrUpdateSec(ctx context.Context, cr *v1alpha1.RepositorysecretsParameters, name string, client client.Client, gh Service) (string, string, error) {
+	encryptedSecret, hash, err := setupEncryptedSecret(ctx, client, cr, name, gh)
+	if err != nil {
+		return "", "", err
+	}
+
+	if _, err := gh.CreateOrUpdateRepoSecret(ctx, cr.Owner, cr.Repository, encryptedSecret); err != nil {
+		return "", "", err
+	}
+
+	sec, _, err := gh.GetRepoSecret(ctx, cr.Owner, cr.Repository, name)
```

Comment:
If you already get the secret response from the API in the `gh.CreateOrUpdateRepoSecret`, why do you discard its value (using `_`) and make another request (`GET` in line 60) right after discarding the response?

I think it would be simpler if you just create `sec` with the returned value from the CreateOrUpdate request.

```suggestion
	if sec, err := gh.CreateOrUpdateRepoSecret(ctx, cr.Owner, cr.Repository, encryptedSecret); err != nil {
		return "", "", err
	}
```

Reply:
The function CreateOrUpdateRepoSecret returns only the status if the secret is created or updated with success. Otherwise, calling GetRepoSecret return more datas, like the time that repository secret updated recently. For more informations, please read [documentation](https://docs.github.com/en/rest/reference/actions#secrets).